mesonbuild/mdist.py

```diff
@@ -189,72 +241,60 @@ def process_submodules(self, src: str, distdir: str) -> None:
             elif status in {'+', 'U'}:
                 handle_dirty_opt(f'Submodule {subpath!r} has uncommitted changes that will not be included in the dist tarball', self.options.allow_dirty)
 
-            self.copy_git(os.path.join(src, subpath), distdir, revision=sha1, prefix=subpath)
+            self.copy_repo(os.path.join(src, subpath), distdir, revision=sha1, prefix=subpath)
 
-    def create_dist(self, archives: T.List[str]) -> T.List[str]:
-        self.process_git_project(self.src_root, self.distdir)
-        for path in self.subprojects.values():
-            sub_src_root = os.path.join(self.src_root, path)
-            sub_distdir = os.path.join(self.distdir, path)
-            if os.path.exists(sub_distdir):
-                continue
-            if is_git(sub_src_root):
-                self.process_git_project(sub_src_root, sub_distdir)
-            else:
-                shutil.copytree(sub_src_root, sub_distdir)
-        self.run_dist_scripts()
-        output_names = []
-        for a in archives:
-            compressed_name = self.distdir + archive_extension[a]
-            shutil.make_archive(self.distdir, a, root_dir=self.dist_sub, base_dir=self.dist_name)
-            output_names.append(compressed_name)
-        windows_proof_rmtree(self.distdir)
-        return output_names
+    def is_repo(self, src_root: str) -> bool:
+        return is_git(src_root)
 
 
 class HgDist(Dist):
+
+    def _check_output(self, cmd: T.Union[list, str]) -> str:
+        if isinstance(cmd, str):
+            cmd = cmd.split()
+        if cmd[0] != 'hg':
+            cmd.insert(0, 'hg')
+        return subprocess.check_output(cmd, cwd=self.src_root, text=True, encoding='utf-8')
+
+    def get_repo_root(self, dir_: str) -> Path:
+        # workaround using meson.build path, see Git above
+        path_rel_meson_build = self._check_output(
+            'status meson.build --all --template {path}'
+        ).strip()
```

Comment:
One line, please.

We have much longer lines in this file already ;) and the max line length is NOT 80 chars. But you could save a few of them anyway by using a variable name such as `buildfile_path` and avoid encoding information about "rel" in there at all. (Not sure what it is supposed to signify.)

Reply:
Done!